src/main/scala/piconot/internal/SurroundingsChecker.scala

```diff
@@ -0,0 +1,40 @@
+package piconot.internal
+import picolib.maze.Maze
+import picolib.semantics._
+
+class SurroundingsChecker(val status:RelativeDescription, 
```

Comment:
Although not as bad, we also had to create several wrapper classes in our own implementation. We felt we could have abstracted the logic from the code more so, similar to what you guys did, but we realized it was probably going to be a huge pain (which it seems like it was, based off of the comments above).